src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -6744,6 +6745,198 @@ def merge(
+        ----------
+        right: other SnowflakeQueryCompiler to merge with.
+        on : label
+            Field name to join on. Must be found in both DataFrames. The data MUST be ordered.
+            Furthermore, this must be a numeric column such as datetimelike, integer, or float.
+            On or left_on/right_on must be given.
+        left_on : label
+            Field name to join on in left DataFrame.
+        right_on : label
+            Field name to join on in right DataFrame.
+        left_index : bool
+            Use the index of the left DataFrame as the join key.
+        right_index : bool
+            Use the index of the right DataFrame as the join key.
+        by : column name or list of column names
+            Match on these columns before performing merge operation.
+        left_by : column name
+            Field names to match on in the left DataFrame.
+        right_by : column name
+            Field names to match on in the right DataFrame.
+        suffixes : 2-length sequence (tuple, list, …)
+            Suffix to apply to overlapping column names in the left and right side, respectively.
+        tolerance: int or Timedelta, optional, default None
+            Select asof tolerance within this range; must be compatible with the merge index.
+        allow_exact_matches : bool, default True
+            If True, allow matching with the same ‘on’ value (i.e. less-than-or-equal-to / greater-than-or-equal-to)
+            If False, don’t match the same ‘on’ value (i.e., strictly less-than / strictly greater-than).
+        direction : ‘backward’ (default), ‘forward’, or ‘nearest’
+            Whether to search for prior, subsequent, or closest matches.
+
+        Returns
+        -------
+        SnowflakeQueryCompiler
+        """
+        if (
+            by
+            or left_by
+            or right_by
+            or left_index
+            or right_index
+            or tolerance
+            or suffixes != ("_x", "_y")
+        ):
+            ErrorMessage.not_implemented(
+                "Snowpark pandas merge_asof method does not currently support parameters "
+                + "'by', 'left_by', 'right_by', 'left_index', 'right_index', "
+                + "'suffixes', or 'tolerance'"
+            )
+        if direction not in ("backward", "forward"):
+            ErrorMessage.not_implemented(
+                "Snowpark pandas merge_asof method only supports directions 'forward' and 'backward'"
+            )
+
+        left_frame = self._modin_frame
+        right_frame = right._modin_frame
+        # Project to Snowpark DataFrame in order to perform ASOF Join
+        left_snowpark_df = (
+            left_frame.ordered_dataframe.to_projected_snowpark_dataframe()
+        )
+        right_snowpark_df = (
+            right_frame.ordered_dataframe.to_projected_snowpark_dataframe()
+        )
+        if on:
+            on_quoted_identifier = quote_name_without_upper_casing(on)
```

Comment:
This will break if the quoted identifier is different from the pandas name, right? (e.g. say we have a series of operations, and the quoted identifier becomes `"pandas_name_xai23"`, which happens sometimes when deduplicating columns) - could we instead use the mapping that the internal frame has from pandas label to quoted identifer to figure out the quoted identifier?

Reply:
Great point, done!